src/Component/TopBanner/TopBanner.tsx

```diff
@@ -1,15 +1,34 @@
 import "./styles.scss";
-import TopBannerPic from '../../assets/pictures/topbanner.svg';
+import TopBannerPic from "../../assets/pictures/topbanner.svg";
+import TopBannerPicLarge from "../../assets/pictures/top_banner.svg";
 import { SearchBanner } from "./SearchBanner";
 import { TOP_BANNER } from "../../resources/content";
-
+import { useState, useEffect } from "react";
 
 export const TopBanner = () => {
+  const [bannerImage, setBannerImage] = useState(TopBannerPic);
+
+  useEffect(() => {
+    const handleResize = () => {
+      if (window.innerWidth > 786) {
+        setBannerImage(TopBannerPicLarge);
+      } else {
+        setBannerImage(TopBannerPic);
+      }
+    };
+    handleResize();
+
+    window.addEventListener("resize", handleResize);
+
+    return () => {
+      window.removeEventListener("resize", handleResize);
+    };
+  }, []);
```

Comment:
try to avoid using event listeners as much as possible
incase you need to change the image depending on the window size - 
either, render both images and work with display:none one of them using css media queries 
or make a div with a background image and change the background image in the css using media queries accordingly

Reply:
fixed.